### PR TITLE
3.x fixes for #3327

### LINKED ISSUE #3327: Deprecation errors in php 8
Seems to be caused by https://php.watch/versions/8.0/deprecate-required-param-after-optional

Twig 2.x

https://travis-ci.org/github/phpmyadmin/phpmyadmin/jobs/686744942#L448

<details><summary>Errors</summary>
<p>

```
1) PhpMyAdmin\Tests\BrowseForeignersTest::testGetHtmlForRelationalFieldSelection
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $arguments follows optional parameter $callable") in "table/browse_foreigners/show_all.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/BrowseForeigners.php:176
test/classes/BrowseForeignersTest.php:206
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $arguments follows optional parameter $callable
vendor/twig/twig/src/Node/Expression/CallExpression.php:116
vendor/twig/twig/src/Node/Expression/TestExpression.php:17
vendor/twig/twig/src/ExpressionParser.php:693
vendor/twig/twig/src/ExpressionParser.php:79
vendor/twig/twig/src/TokenParser/IfTokenParser.php:36
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/BrowseForeigners.php:176
test/classes/BrowseForeignersTest.php:206
2) PhpMyAdmin\Tests\CentralColumnsTest::testGetHtmlForMain
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "database/central_columns/main.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/CentralColumns.php:1212
test/classes/CentralColumnsTest.php:652
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/CentralColumns.php:1212
test/classes/CentralColumnsTest.php:652
3) PhpMyAdmin\Tests\Config\FormDisplayTemplateTest::testDisplayTabsTop
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "list/unordered.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
test/classes/Config/FormDisplayTemplateTest.php:78
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
test/classes/Config/FormDisplayTemplateTest.php:78
4) PhpMyAdmin\Tests\Config\PageSettingsTest::testShowGroupBrowse
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "list/unordered.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
libraries/classes/Config/FormDisplay.php:336
libraries/classes/Config/PageSettings.php:168
libraries/classes/Config/PageSettings.php:87
libraries/classes/Config/PageSettings.php:206
test/classes/Config/PageSettingsTest.php:51
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
libraries/classes/Config/FormDisplay.php:336
libraries/classes/Config/PageSettings.php:168
libraries/classes/Config/PageSettings.php:87
libraries/classes/Config/PageSettings.php:206
test/classes/Config/PageSettingsTest.php:51
5) PhpMyAdmin\Tests\Config\PageSettingsTest::testGetNaviSettings
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "list/unordered.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
libraries/classes/Config/FormDisplay.php:336
libraries/classes/Config/PageSettings.php:168
libraries/classes/Config/PageSettings.php:87
libraries/classes/Config/PageSettings.php:222
test/classes/Config/PageSettingsTest.php:85
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Config/FormDisplayTemplate.php:105
libraries/classes/Config/FormDisplay.php:336
libraries/classes/Config/PageSettings.php:168
libraries/classes/Config/PageSettings.php:87
libraries/classes/Config/PageSettings.php:222
test/classes/Config/PageSettingsTest.php:85
6) PhpMyAdmin\Tests\Controllers\Database\StructureControllerTest::testDisplayTableList
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "database/structure/table_header.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/Database/StructureController.php:1131
test/classes/Controllers/Database/StructureControllerTest.php:592
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/Database/StructureController.php:1131
test/classes/Controllers/Database/StructureControllerTest.php:592
7) PhpMyAdmin\Tests\Controllers\Server\DatabasesControllerTest::testIndexAction
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "server/databases/index.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/AbstractController.php:45
libraries/classes/Controllers/Server/DatabasesController.php:178
test/classes/Controllers/Server/DatabasesControllerTest.php:67
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/AbstractController.php:45
libraries/classes/Controllers/Server/DatabasesController.php:178
test/classes/Controllers/Server/DatabasesControllerTest.php:67
8) PhpMyAdmin\Tests\Controllers\Server\VariablesControllerTest::testIndex
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "server/variables/index.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/AbstractController.php:45
libraries/classes/Controllers/Server/VariablesController.php:93
test/classes/Controllers/Server/VariablesControllerTest.php:92
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Controllers/AbstractController.php:45
libraries/classes/Controllers/Server/VariablesController.php:93
test/classes/Controllers/Server/VariablesControllerTest.php:92
9) PhpMyAdmin\Tests\Database\DesignerTest::testGetHtmlForEditOrDeletePages
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "database/designer/edit_delete_pages.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Database/Designer.php:67
test/classes/Database/DesignerTest.php:137
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Database/Designer.php:67
test/classes/Database/DesignerTest.php:137
10) PhpMyAdmin\Tests\Database\DesignerTest::testGetHtmlForPageSaveAs
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "database/designer/page_save_as.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Database/Designer.php:85
test/classes/Database/DesignerTest.php:166
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Database/Designer.php:85
test/classes/Database/DesignerTest.php:166
11) PhpMyAdmin\Tests\Display\ExportTest::testGetHtmlForOptions
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "display/export/options_format.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Display/Export.php:256
libraries/classes/Display/Export.php:566
test/classes/Display/ExportTest.php:145
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Display/Export.php:256
libraries/classes/Display/Export.php:566
test/classes/Display/ExportTest.php:145
12) PhpMyAdmin\Tests\HeaderTest::testEnable
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "console/display.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Console.php:137
libraries/classes/Header.php:446
test/classes/HeaderTest.php:71
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Console.php:137
libraries/classes/Header.php:446
test/classes/HeaderTest.php:71
13) PhpMyAdmin\Tests\HeaderTest::testSetBodyId
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "console/display.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Console.php:137
libraries/classes/Header.php:446
test/classes/HeaderTest.php:86
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Console.php:137
libraries/classes/Header.php:446
test/classes/HeaderTest.php:86
14) PhpMyAdmin\Tests\NormalizationTest::testGetHtmlForCreateNewColumn
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "columns_definitions/table_fields_definitions.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Normalization.php:202
test/classes/NormalizationTest.php:150
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Normalization.php:202
test/classes/NormalizationTest.php:150
15) PhpMyAdmin\Tests\RelationTest::testPMAGetRelationsParam
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "list/unordered.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Relation.php:426
test/classes/RelationTest.php:103
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/ForTokenParser.php:52
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Relation.php:426
test/classes/RelationTest.php:103
16) PhpMyAdmin\Tests\SqlQueryFormTest::testPMAGetHtmlForSqlQueryFormInsert
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "sql/query.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/SqlQueryForm.php:127
test/classes/SqlQueryFormTest.php:110
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/SqlQueryForm.php:127
test/classes/SqlQueryFormTest.php:110
17) PhpMyAdmin\Tests\SqlQueryFormTest::testPMAGetHtmlForSqlQueryForm
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "sql/query.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/SqlQueryForm.php:127
test/classes/SqlQueryFormTest.php:175
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/SqlQueryForm.php:127
test/classes/SqlQueryFormTest.php:175
18) PhpMyAdmin\Tests\TrackingTest::testGetHtmlForMain
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "table/tracking/main.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Tracking.php:189
test/classes/TrackingTest.php:214
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/TokenParser/IfTokenParser.php:39
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Tracking.php:189
test/classes/TrackingTest.php:214
19) PhpMyAdmin\Tests\TwoFactorTest::testKey
Twig\Error\SyntaxError: An exception has been thrown during the compilation of a template ("Required parameter $lineno follows optional parameter $only") in "login/twofactor/key.twig".
vendor/twig/twig/src/Environment.php:600
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Plugins/TwoFactor/Key.php:128
libraries/classes/TwoFactor.php:218
test/classes/TwoFactorTest.php:205
Caused by
PHPUnit\Framework\Error\Deprecated: Required parameter $lineno follows optional parameter $only
vendor/twig/twig/src/Node/IncludeNode.php:25
vendor/twig/twig/src/TokenParser/IncludeTokenParser.php:33
vendor/twig/twig/src/Parser.php:185
vendor/twig/twig/src/Parser.php:98
vendor/twig/twig/src/Environment.php:563
vendor/twig/twig/src/Environment.php:595
vendor/twig/twig/src/Environment.php:408
vendor/twig/twig/src/Environment.php:381
vendor/twig/twig/src/Environment.php:359
libraries/classes/Template.php:101
libraries/classes/Template.php:134
libraries/classes/Plugins/TwoFactor/Key.php:128
libraries/classes/TwoFactor.php:218
test/classes/TwoFactorTest.php:205
```

</p>
</details>

### PR DESCRIPTION
**Pull request by williamdes**
Fixes: #3327

WIP

### CODE REVIEW

File: src/Node/Expression/CallExpression.php [hunk 1]
```diff
@@ -112,7 +112,7 @@ protected function compileArguments(Compiler $compiler, $isArray = false): void
         $compiler->raw($isArray ? ']' : ')');
     }
 
-    protected function getArguments($callable = null, $arguments)
+    protected function getArguments($callable = null, $arguments = [])
```
Comment (stof) [inline src/Node/Expression/CallExpression.php:115]: I'd rather remove `= null` here (I guess that there was a typehint at some point justifying the `= null` to make the typehint nullable, and it was removed later)

File: src/Node/ForNode.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ class ForNode extends Node
 {
     private $loop;
 
-    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, $ifexpr = null, Node $body, Node $else = null, int $lineno, string $tag = null)
+    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, $ifexpr, Node $body, ?Node $else, int $lineno, ?string $tag = null)
```
Comment (fabpot) [inline src/Node/ForNode.php:28]: What about adding default values instead? That way, we could merge it in 1.x.

Comment (williamdes) [reply src/Node/ForNode.php:28]: It was refused in #3327 ~~but I will do that tomorrow
Edit: Sometime soon~~

**Comment by fabpot** [comment]
I'm going to submit a PR on the 1.x branch for everything that can be done there.

**Comment by williamdes** [comment]
> I'm going to submit a PR on the 1.x branch for everything that can be done there.

Thank you!
It will help me see how to fix the issues left to be fixed here when I can find time after work

**Comment by fabpot** [comment]
I've done the work for 1.x. Merging 1.x into 2.x and then 3.x will fix all issues.

**Comment by williamdes** [comment]
You rock !
Thank you !
Could you post a list of commits here for the record, please ?

**Comment by xabbuh** [comment]
@williamdes see #3331

**Comment by williamdes** [comment]
Thank you @xabbuh I will link everything with it for he record

**Pull request closed without merging**